pkg/storage/image.go

```diff
@@ -589,25 +590,35 @@ func GetImageService(ctx context.Context, store storage.Store, defaultTransport
 		}
 	}
 
-	seenRegistries := make(map[string]bool, len(registries))
-	cleanRegistries := []string{}
-	for _, r := range registries {
-		if seenRegistries[r] {
-			continue
-		}
-		cleanRegistries = append(cleanRegistries, r)
-		seenRegistries[r] = true
-	}
-
 	is := &imageService{
 		store:                 store,
 		defaultTransport:      defaultTransport,
 		indexConfigs:          make(map[string]*indexInfo),
 		insecureRegistryCIDRs: make([]*net.IPNet, 0),
-		registries:            cleanRegistries,
 		imageCache:            make(map[string]imageCacheItem),
 		ctx:                   ctx,
 	}
+	if len(registries) != 0 {
+		seenRegistries := make(map[string]bool, len(registries))
+		cleanRegistries := []string{}
+		for _, r := range registries {
+			if seenRegistries[r] {
+				continue
+			}
+			cleanRegistries = append(cleanRegistries, r)
+			seenRegistries[r] = true
+		}
+
+		is.registries = cleanRegistries
+	} else {
+		systemRegistries, err := sysregistriesv2.FindUnqualifiedSearchRegistries(sc)
+		if err != nil {
+			return nil, err
+		}
+		for _, r := range systemRegistries {
+			is.registries = append(is.registries, r.URL)
```

Comment:
I would like recommend to rename `registries` from the `imageService` structure to something like `unqualifiedSearchRegistries`.

Reply:
done, thanks!